### ISSUE TITLE
[BUG] After updating the mod, entities stay completely still and their nametags say entity.minecolonies.citizen . After restarting the world, the colony is empty and the building hut blocks can't be interacted with

### ISSUE THREAD

**Issue by 147-258-369** [ldtteam/minecolonies#9769]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [X] Single Player
- [ ] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

1.20.1-1.1.468-BETA

### Structurize Version

1.20.1-1.0.686-BETA 

### Current Behavior

After updating the game to the latest version, all the colonists' skins reset and their nametags had their placeholder names "entity.minecolonies.citizen", I reloaded the world, and the colony was empty, and I could only make a new colony from the townhall. The game didn't register the buildings around the townhall, and that's how I got the error in the second picture. Before this, the Builder and Forester AI kept glitching; the Builder would commonly stop building, and I would have to fire them and cancel their build and reset by rehiring them and restarting their build upgrading. Even after firing and rehiring the forester, they wouldn't chop down any trees in their zone (I know the forester bug is an issue here already, but I'm saying this in case this has to do with what I just ran into) (Oh and just in case, I have dynamic trees for 1.20.1-BETA2 . I don't know if that mod's Beta has to do with the forester's problems)
![2024-02-04_15 44 36](https://github.com/ldtteam/minecolonies/assets/158307975/585fad4f-21fa-40e8-ab57-ae8f56868c9e)
![2024-02-04_16 23 43](https://github.com/ldtteam/minecolonies/assets/158307975/257717d7-61aa-410d-8117-9499adc182c8)

### Expected Behavior

I thought I just had to restart the world; I exited the game, restarted it and after restarting it, that's when all the colonists disappeared.

### Reproduction Steps

1. Update to the latest version of (2/4/2024)
2. Load a world
3. restart the game and load the world again
![2024-02-04_16 23 43](https://github.com/ldtteam/minecolonies/assets/158307975/5c96758d-9622-4031-b54e-c66260fdb65f)

### Logs

https://gist.github.com/147-258-369/8908ed57d70772bff600fbb6a9fa8b80

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.
[2024-02-04-3.log.gz](https://github.com/ldtteam/minecolonies/files/14158571/2024-02-04-3.log.gz)
[2024-02-04-4.log.gz](https://github.com/ldtteam/minecolonies/files/14158572/2024-02-04-4.log.gz)
[2024-02-04-5.log.gz](https://github.com/ldtteam/minecolonies/files/14158573/2024-02-04-5.log.gz)
[2024-02-04-6.log.gz](https://github.com/ldtteam/minecolonies/files/14158574/2024-02-04-6.log.gz)
[2024-02-04-7.log.gz](https://github.com/ldtteam/minecolonies/files/14158575/2024-02-04-7.log.gz)
[2024-02-04-1.log.gz](https://github.com/ldtteam/minecolonies/files/14158576/2024-02-04-1.log.gz)
[2024-02-04-2.log.gz](https://github.com/ldtteam/minecolonies/files/14158577/2024-02-04-2.log.gz)
[latest.log](https://github.com/ldtteam/minecolonies/files/14158578/latest.log)
[debug-3.log.gz](https://github.com/ldtteam/minecolonies/files/14158580/debug-3.log.gz)
[debug-4.log.gz](https://github.com/ldtteam/minecolonies/files/14158581/debug-4.log.gz)
[debug-5.log.gz](https://github.com/ldtteam/minecolonies/files/14158582/debug-5.log.gz)
[debug-1.log.gz](https://github.com/ldtteam/minecolonies/files/14158583/debug-1.log.gz)
[debug-2.log.gz](https://github.com/ldtteam/minecolonies/files/14158584/debug-2.log.gz)

Here are all the logs and debug files just in case.

**Comment by Raycoms** [comment]
Got a fix building right now.

**Comment by Raycoms** [comment]
Once approved: https://legacy.curseforge.com/minecraft/mc-mods/minecolonies/files/5081478

**Comment by AliceDog98** [comment]
I am having the same issue with the 1.20 version. Both of my builders are entity.minecolonies.citizen so I can't get their inventories, or recall them.  Also, I am having the same issue with the game saying "This block is missing its respective building, try restarting or loading a backup." I fixed the block by clicking on the block again, and it would work normally. I saw on the discord server that all I have to do is put in a command and back up the colony but I am worried all my colony will go back to level one.
